src/hooks/useDisplayedWallets.ts

```diff
@@ -1,38 +1,72 @@
 import { StatSigFlags } from '@/constants/statsig';
-import { WalletType } from '@/constants/wallets';
+import { KEPLR_MIPD_RDNS, OKX_MIPD_RDNS, PHANTOM_MIPD_RDNS } from '@/constants/wallets';
 
 import { isTruthy } from '@/lib/isTruthy';
+import { ConnectorType, WalletInfo, WalletType } from '@/lib/wallet/types';
 
+import { useMipdInjectedWallets } from './useMipdInjectedWallets';
 import { useStatsigGateValue } from './useStatsig';
 
-export const useDisplayedWallets = () => {
+export const useDisplayedWallets = (): WalletInfo[] => {
   const keplrEnabled = useStatsigGateValue(StatSigFlags.ffEnableKeplr);
+  const injectedWallets = useMipdInjectedWallets();
+
+  return useMemo(() => {
+    const phantomDetected =
+      injectedWallets.findIndex((wallet) => wallet.detail.info.rdns === PHANTOM_MIPD_RDNS) !== -1;
 
-  const displayedWallets = [
-    WalletType.MetaMask,
+    const okxDetected =
+      injectedWallets.findIndex((wallet) => wallet.detail.info.rdns === OKX_MIPD_RDNS) !== -1;
 
-    keplrEnabled && WalletType.Keplr,
+    return [
+      ...injectedWallets
+        // Remove Keplr EVM support since Keplr Cosmos is supported
+        // Remove Phantom EVM support, but enable Phantom Solana support based on EIP-6963 detection
+        .filter(
+          (wallet) =>
+            wallet.detail.info.rdns !== PHANTOM_MIPD_RDNS &&
+            wallet.detail.info.rdns !== KEPLR_MIPD_RDNS
+        )
+        .map(
+          (wallet) =>
+            ({
+              connectorType: ConnectorType.Injected,
+              icon: wallet.detail.info.icon,
+              name: wallet.detail.info.name,
+              rdns: wallet.detail.info.rdns,
+            }) as WalletInfo
+        ),
 
-    Boolean(isTestnet || isDev) && WalletType.Phantom,
+      (isTestnet || isDev) &&
+        phantomDetected && {
+          connectorType: ConnectorType.PhantomSolana,
+          name: WalletType.Phantom,
+        },
 
-    WalletType.WalletConnect2,
+      keplrEnabled && {
+        connectorType: ConnectorType.Cosmos,
+        name: CosmosWalletType.KEPLR,
+      },
 
-    WalletType.CoinbaseWallet,
+      { connectorType: ConnectorType.WalletConnect, name: WalletType.WalletConnect2 },
 
-    WalletType.OkxWallet,
-    // Hide these wallet options until they can be properly tested on mainnet
-    // WalletType.ImToken,
-    // WalletType.Rainbow,
-    // WalletType.TrustWallet,
-    // WalletType.HuobiWallet,
-    // WalletType.BitKeep,
-    // WalletType.Coin98,
+      { connectorType: ConnectorType.Coinbase, name: WalletType.CoinbaseWallet },
 
-    Boolean(import.meta.env.VITE_PRIVY_APP_ID) && WalletType.Privy,
+      // No need to special-case an OKX WalletConnect option if the OKX extension wallet is already detected.
+      // Note that OKX mobile app users can still connect through the generic WalletConnect option
+      !okxDetected && { connectorType: ConnectorType.WalletConnect, name: WalletType.OkxWallet },
```

Comment:
I wonder if we can do the same with CoinbaseWallet to prevent smart wallet. So if we don't detect CoinbaseWallet extension, we can force wallet connect, and avoid the smart wallet this way.